irida_import/irida_import.py

```diff
@@ -181,8 +183,19 @@ def get_sample_file(self, file_dict):
         resource = file_dict
         name = resource['fileName']
         path = resource['file']
-
-        return SampleFile(name, path)
+        links = file_dict['links']
+        for link in links:
+            if link['rel'] == 'self':
+                href = link['href']
+                break
+        upload_sha_256 = None
+        file_size = None
+        if 'uploadSha256' in resource.keys():
+            upload_sha_256 = resource['uploadSha256']
+        if 'fileSizeBytes' in resource.keys():
+            file_size = resource['fileSizeBytes']
+
+        return SampleFile(name, path, href, file_size, upload_sha_256)
```

Comment:
Here's another line where explicitly stating the arguments would be good because there are 5 arguments.

Reply:
Updated in 97c187bc8ab0716b21f78c299f47c20ad4f1ae00